### ISSUE TITLE
Error: inbound message key "chatId" for transport universal is empty

### ISSUE THREAD

**Issue by Eeems** [guidone/node-red-contrib-chatbot#250]
I'm getting the following error:
>  Error: inbound message key "chatId" for transport universal is empty
> 
> See here: https://github.com/guidone/node-red-contrib-chatbot/wiki/Universal-Connector-node for an NaN

The second line of the error seems to be from two `+` symbols on [lib/chat-platform/chat-platform.js#L106](https://github.com/guidone/node-red-contrib-chatbot/blob/master/lib/chat-platform/chat-platform.js#L106)

Here is my flow:
![image](https://user-images.githubusercontent.com/1176979/46301696-50ee5c00-c564-11e8-8549-6a876f79bc91.png)

The contents of my Setup block
```javascript
return {
    payload: {
        type: 'message',
        content: msg.payload,
        chatId: msg.to,
        userId: msg.to,
        messageId: msg._msgid
    }
};
```

The Extend block:
```javascript
node.chat.onChatId(message => message.originalMessage.chatId);
node.chat.onMessageId(message => message.originalMessage.messageId);
node.chat.in(message => new Promise((resolve, reject) => {
    resolve(message);
}));
node.chat.out('message', message => {
  return new Promise((resolve, reject) => {
     message.payload = message.payload.content;
     message.topic = message.originalMessage.to;
     if(message.topic == "PRIVMSG"){
         message.topic = message.originalMessage.from;
     }
     resolve(message);
  });    
});
```

**Comment by guidone** [comment]
Hi @Eeems the first problem is onChatId(...), it should be

```
node.chat.onChatId(message => message.payload.chatId);
node.chat.onMessageId(message => message.payload.messageId);
```

When onChatId and onMessageId are executed _originalMessage_ is not ready yet, perhaps the documentation is wrong, I'll fix it.

The second problem is here

```javascript
node.chat.in(message => new Promise((resolve, reject) => {
    resolve(message);
}));
```

You have to assign a _type_ to the incoming message otherwise the message will never go out to the output pin and will be considered "noise", so for example

```
node.chat.in(message => new Promise((resolve, reject) => {
    message.content.type = 'message';
    resolve(message);
}));
```

Third problem: the universal connector answer badly if the chatId is null (with this NaN), it should answer with a more meaningful error, I'll fix that.

Thanks for the feedback

**Comment by Eeems** [comment]
Yeah I wrote all my code based on the following wiki pages and they will need to be updated:
https://github.com/guidone/node-red-contrib-chatbot/wiki/Universal-Connector-node
https://github.com/guidone/node-red-contrib-chatbot/wiki/Extend-node

Thanks for the quick reply!

**Comment by guidone** [comment]
Hang tight @Eeems I'm going to fix the example and some other fixes about the universal connector (tonight)

**Comment by guidone** [comment]
Fixed on `0.15.4`
